libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1006,50 +1007,54 @@ void SMTEncoder::arrayAssignment()
 	m_arrayAssignmentHappened = true;
 }
 
-void SMTEncoder::arrayIndexAssignment(Expression const& _expr, smtutil::Expression const& _rightHandSide)
+void SMTEncoder::indexOrMemberAssignment(Expression const& _expr, smtutil::Expression const& _rightHandSide)
```

Comment:
Is it maybe better to split this one to two functions? Or better, is there a reason for structs and arrays using same function?

Reply:
Yes, the function is recursive for cases like `s.a[2].b[3]...`

Reply:
Sorry, not recursive, the logic is just inside a single loop